research/src/components/checkbox-anatomy.js

```diff
@@ -0,0 +1,18 @@
+import React from 'react'
+import './anatomy.css'
+
+export default () => {
+  return (
+    <div className="component-anatomy">
+      <host>
+        <part name="label">
+          <slot>{'Default'}</slot>
+        </part>
+        <part name="checkbox">
+          <part name="checked-indicator" />
```

Comment:
I'm intrigued at the need for different parts. Is this to inform that there are two states of which can both be on/off? Cause ultimately this could just be a series of icons that have their visibility toggled upon state change, no?

Reply:
I think thats a good point. Another point that was brought up was that, being different elements, it would make it much harder to *transition* between visual treatments of checked vs indeterminate.

But yes, the original idea was that because `checked` and `indeterminate` are distinct states, there could be mechanisms to provide indicators for each. Internally, default behavior of the control would show indeterminate when indeterminate, and when indeterminate is false, would show checked when checked. Exposing a single part / slot w/ a set of child icons would require more work by the app-author to indicate which should be shown when.

Reply:
As you've suggested I've created an [issue ](https://github.com/WICG/open-ui/issues/93) to discuss.